lib/iris/_lazy_data.py

```diff
@@ -126,6 +126,34 @@ def as_concrete_data(data):
     return data
 
 
+def lazy_masked_fill_value(data):
```

Comment:
You're going to have to try harder than that to convince me 😉 

I guess that...

* I only care about fill values if I have masked data,
* so 'masked' could be omitted in that sense, but
* clarity is a good thing – it's in the masked data case that I want to retrieve the fill value.

So I'm not currently buying into a rebrand! What do you see as the benefit of doing so?